scripts/telegram_ros_bridge

```diff
@@ -65,19 +67,27 @@ class TelegramROSBridge(object):
     def _telegram_start_callback(self, _, update):
         """
         Called when a telegram user sends the '/start' event to the bot, using this event, the bridge can be connected
-        to a specific conversation
+        to a specific conversation.
+        This checks whether the user is on the whitelist if the whitelist is a list and allows any user if whitelist is False
+
         :param update: Received update event that holds the chat_id and message data
         """
-        if self._telegram_chat_id is not None and self._telegram_chat_id != update.message.chat_id:
-            rospy.logwarn("Changing to different chat_id!")
-            self._telegram_updater.bot.send_message(self._telegram_chat_id,
-                                                    "Lost ROS bridge connection to this chat_id (somebody took over)")
-
-        self._telegram_chat_id = update.message.chat_id
-
-        rospy.loginfo("Starting telegram ROS bridge for chat id {}".format(self._telegram_chat_id))
-        update.message.reply_text(
-            "Telegram ROS bridge initialized, only replying to chat_id {} (current). Type /stop to disconnect".format(self._telegram_chat_id))
+        
+        # Whitelist defaults to False to disable it and allow anyone. To enable the whitelist, set it to an (empty) list
+        whitelist = rospy.get_param(WHITELIST, False)
+        if whitelist == False or isinstance(whitelist, list) and update.message.from_user.id in whitelist:
+            if self._telegram_chat_id is not None and self._telegram_chat_id != update.message.chat_id:
+                rospy.logwarn("Changing to different chat_id!")
+                self._telegram_updater.bot.send_message(self._telegram_chat_id,
+                                                        "Lost ROS bridge connection to this chat_id (somebody took over)")
+            self._telegram_chat_id = update.message.chat_id
+
+            rospy.loginfo("Starting telegram ROS bridge for chat id {}".format(self._telegram_chat_id))
+            update.message.reply_text(
+                "Telegram ROS bridge initialized, only replying to chat_id {} (current)".format(self._telegram_chat_id))
+        else:
+            rospy.logwarn("Discarding message. User {} not whitelisted".format(update.message.from_user))
+            update.message.reply_text("You (chat id {}) are not authorized to chat with this bot".format(update.message.from_user['id']))
```

Comment:
In this part `update.message.from_user.id`, `update.message.chat_id` and `update.message.from_user['id']` are used. Are these the same or different. In case these are the same, please use one of them.

Reply:
- `update.message.from_user.id` is the [sender](https://python-telegram-bot.readthedocs.io/en/latest/telegram.message.html#telegram.Message.from_user) of the message
- `update.message.chat_id` is [a shortcut for `Chat.id` ](https://python-telegram-bot.readthedocs.io/en/stable/telegram.message.html#telegram.Message.chat_id)
- `update.message.from_user['id']` is indeed the same as  `update.message.from_user.id` from what I can find. 

- [`update`](https://python-telegram-bot.readthedocs.io/en/latest/telegram.update.html#telegram-update)
- [`message`](https://python-telegram-bot.readthedocs.io/en/latest/telegram.message.html#telegram.Message)
- `from_user`

I cannot find in the docs that chat_id and user ID are equivalent (although they seem that way) so I'll leave that as is.